### PR TITLE
engage siteAdapters, err details to console

### PR DESCRIPTION
**Pull request by WardCunningham**
Here we use LINEUP to diagram pages from both http and https sites. With this now working we can consider how other algorithmic markup can see deeper into the federation.

image

### CODE REVIEW

**Comment by WardCunningham** [comment]
Remove MEHAFFY & diagram()
Add Redirect argument to DOT keyword.
Add new about pages including MEHAFFY replacement.

**Comment by WardCunningham** [comment]
I've changed the redirect syntax to be more readable and unambiguous at the same time by signaling redirect with `DOT FROM`.

I had overloaded the DOT keyword in a way that `DOT graph` could have two meanings. At first I didn't consider this a serious problem because a page named Graph would have to contain a Graphviz plugin before the second meaning would be unavailable. But why be so clever. 

image

**Comment by paul90** [comment]
As discussed, this looks good for publishing a `next` tagged version from this branch.

**Comment by WardCunningham** [comment]
This fetch fails on ghost page.

image

Debugging with "Pause on Exception" we find this code in jQuery-migrate that is trying to catch this error but not getting the null that it expected.

image

**Comment by WardCunningham** [comment]
Lineup Diagram once again readily retrieved by searching for it. Here is my rather complicated test case that had be caught in an edit-publish-debug loop.

image

**Comment by WardCunningham** [comment]
We now accept DOT FROM with or without a \n.
We report slightly better error messages when the redirect doesn't work.

image

**Comment by WardCunningham** [comment]
Add widely useful Lineup Diagram to about pages.

image

**Comment by WardCunningham** [comment]
We now include a brief enumeration of RegExp Metacharacters which is cited in the More About Algorithmic Markup page. This would otherwise be a broken link on most sites.

image

File: client/graphviz.js [hunk 1]
```diff
@@ -15,11 +15,28 @@
...
-      return diagram ($item, item)
-    } else if (m = text.match(/^DOT ((strict )?(di)?graph)\n/)) {
+    var text = item.text
+    if (m = text.match(/^DOT FROM ([a-z0-9-]+)($|\n)/)) {
+      let site = $item.parents('.page').data('site')||location.host
+      let slug = m[1]
+      var page = null
+      try {
+        page = await wiki.site(site).get(`${slug}.json`, (err, page) => page)
+      } catch (err) {}
```
Comment (dobbs) [inline client/graphviz.js]: does this want to log the error to the console?
```suggestion
      } catch (err) {console.error('failed redirect', site, slug, err)}
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,9 @@
 {
   "name": "wiki-plugin-graphviz",
-  "version": "0.2.2",
+  "version": "0.3.0-7",
+  "publishConfig": {
```
Comment (dobbs) [inline package.json]: Reminder that we'll want to change these details as we publish the final 3.0 version. But that will come after we've merged this PR.

**Pull request merged**